src/vm/hints/find_element_hint.rs

```diff
@@ -0,0 +1,558 @@
+                .iter()
+                .map(|s| s.to_string())
+                .collect(),
+            ids.into_keys().collect(),
+        ));
+    };
+
+    let (array_ptr_addr, elm_size_addr, n_elms_addr, index_addr, key_addr) = if let (
+        Ok(Some(array_ptr_addr)),
+        Ok(Some(elm_size_addr)),
+        Ok(Some(n_elms_addr)),
+        Ok(Some(index_addr)),
+        Ok(Some(key_addr)),
+    ) = (
+        get_address_from_reference(
+            array_ptr_ref,
+            &vm.references,
+            &vm.run_context,
+            vm,
+            hint_ap_tracking,
+        ),
+        get_address_from_reference(
+            elm_size_ref,
+            &vm.references,
+            &vm.run_context,
+            vm,
+            hint_ap_tracking,
+        ),
+        get_address_from_reference(
+            n_elms_ref,
+            &vm.references,
+            &vm.run_context,
+            vm,
+            hint_ap_tracking,
+        ),
+        get_address_from_reference(
+            index_ref,
+            &vm.references,
+            &vm.run_context,
+            vm,
+            hint_ap_tracking,
+        ),
+        get_address_from_reference(
+            key_ref,
+            &vm.references,
+            &vm.run_context,
+            vm,
+            hint_ap_tracking,
+        ),
+    ) {
+        (
+            array_ptr_addr,
+            elm_size_addr,
+            n_elms_addr,
+            index_addr,
+            key_addr,
+        )
+    } else {
+        return Err(VirtualMachineError::FailedToGetIds);
+    };
+
+    match (
+        vm.memory.get(&array_ptr_addr),
+        vm.memory.get(&elm_size_addr),
+        vm.memory.get(&n_elms_addr),
+        vm.memory.get(&index_addr),
+        vm.memory.get(&key_addr),
+    ) {
+        (
+            Ok(_),
+            Ok(Some(maybe_rel_elm_size)),
+            Ok(Some(maybe_rel_n_elms)),
+            Ok(_),
+            Ok(Some(maybe_rel_key)),
+        ) => {
+            for (name, builtin) in &vm.builtin_runners {
+                //Check that range_check_builtin is present
+                let _builtin = match builtin.as_any().downcast_ref::<RangeCheckBuiltinRunner>() {
+                    Some(b) => b,
+                    None => return Err(VirtualMachineError::NoRangeCheckBuiltin),
+                };
+
+                if name == &"range_check".to_string() {
+                    let elm_size = if let MaybeRelocatable::Int(ref elm_size) = maybe_rel_elm_size {
+                        elm_size
+                    } else {
+                        return Err(VirtualMachineError::ExpectedInteger(
+                            maybe_rel_elm_size.clone(),
+                        ));
+                    };
+
+                    if !elm_size.is_positive() {
+                        return Err(VirtualMachineError::ValueOutOfRange(elm_size.clone()));
+                    }
+
+                    if let Some(find_element_index_value) = vm.find_element_index.clone() {
```

Comment:
Why not a reference?

Reply:
`find_element_index` should probably be part of an exec scope.

Reply:
resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/294/commits/24979a3c5067e6b1097e6e7555da60aee8639164